feat(terminal): respond to OSC background and foreground request (#17197)
neovim/neovim: runtime/doc/news.txt

• Terminal buffers emit a |TermRequest| autocommand event when the child
  process emits an OSC or DCS control sequence.

> • The terminal emulator responds to OSC background and foreground requests.
>   See |default-autocmds| for details (the TermRequest item under the
>   nvim_terminal group).

Reviewer: ```suggestion
• Terminal buffers respond to OSC background and foreground requests |default-autocmds|
```

Author: Done (28bf7b9681008fbd7b2f2722252c361e1483ff9c).
